### PR TITLE
Add test for restrictions of Union

### PR DESCRIPTION
**Pull request by matriv**

### CODE REVIEW

**Comment by matriv** [comment]
@mfussenegger please review

File: sql/src/test/java/io/crate/integrationtests/UnionIntegrationTest.java [hunk 1]
```diff
@@ -80,6 +81,17 @@ public void testUnionAllWithOrderBy() throws Exception {
                                                                     "small\n"));
     }
 
+    @Test
+    public void testUnionAsPartOfJoin() {
```
Comment (mfussenegger) [inline sql/src/test/java/io/crate/integrationtests/UnionIntegrationTest.java]: please do this as unit / analyzer test

Comment (mfussenegger) [inline sql/src/test/java/io/crate/integrationtests/UnionIntegrationTest.java]: yes

**Comment by matriv** [comment]
@mfussenegger Please check again

**Pull request merged**